### PR TITLE
Easier and more efficient maintenance of hyperformula-demos

### LINKED ISSUE #976: Improve the maintenance of hyperformula-demos
### Description

Before each release we want to make sure that all the demos works fine and are embedded in the docs correctly
- [x] add readme to the repo
- [x] create a script to update HF version in all the demos
- [x] create a script or an instruction to run verdaccio server locally with the HF release candidate and serve it for the demos
- [x] demos should be tested in environment as close to codesandbox as possible
- [x] for each demo:
  - make it easier what HF version is being used (print version to the console?)
  - configure the node version used by codesandbox
  - `npm run start` should work seamlessly
- [x] remove `master` branch
- [x] set `develop` as a default branch
- [x] update HyperFormula release instructions

### Links

- https://github.com/handsontable/hyperformula-demos
- https://kb.handsontable.com/display/PH/HyperFormula+release+checklist (internal resource)
- https://handsoncode.slack.com/archives/C031GBX62SE/p1649919290421599 (internal discussion)
- https://handsoncode.slack.com/archives/C031GBX62SE/p1649681867862009 (internal discussion)
- #963

### PR DESCRIPTION
**Pull request by sequba**
### Description of the change

As described in https://github.com/handsontable/hyperformula/issues/976:
> - [x] add readme to the repo
> - [x] create a script to update HF version in all the demos
> - [x] create a script or an instruction to run verdaccio server locally with the HF release candidate and serve it for the demos
> - [x] demos should be tested in environment as close to codesandbox as possible
> - [x] for each demo:
>   - make it easier what HF version is being used (print version to the console?)
>   - configure the node version used by codesandbox
>   - `npm run start` should work seamlessly

TODO:
> - [x] remove `master` branch
> - [x] set `develop` as a default branch
> - [x] update HyperFormula release instructions

### How has this been tested?

Locally. The upcoming release will be the real test for these changes.

### Related issue(s):
Fixes https://github.com/handsontable/hyperformula/issues/976

### CODE REVIEW

**Comment by sequba** [comment]
TODO
- [ ] remove master branch 
- [ ] make develop default branch

File: README.md [hunk 1]
```diff
@@ -1 +1,102 @@
-# hyperformula-demos
\ No newline at end of file
+# hyperformula-demos
+
+This repository contains demos and usage examples for the [HyperFormula](https://github.com/handsontable/hyperformula) library. Some of them are embeded in the [HyperFormula documentation](https://handsontable.github.io/hyperformula/).
```
Comment (warpech) [inline README.md]: embeded -> embedded

File: README.md [hunk 2]
```diff
@@ -1 +1,102 @@
...
+
+## Branches
+
+We keep:
+- **develop** branch
+- version branches e.g. **1.0.x**, **1.1.x**, **2.0.x**
+
+The demos on each version branch works with the corresponding version of HyperFormula.
```
Comment (warpech) [inline README.md]: works -> work

File: README.md [hunk 3]
```diff
@@ -1 +1,102 @@
...
+
+## Running the demos
+
+Each directory contains one demo. 
+
+*Make sure to use the right version of the demos. E.g. for **hyperformula@1.3.1** you should use the code from branch **1.3.x**.*
+
+### In CodeSandbox.io
+
+The demos are embeded in HyperFormula documentation using CodeSandbox.io so this should be treated as the primary execution environment for the demos.
```
Comment (warpech) [inline README.md]: embeded -> embedded

File: README.md [hunk 4]
```diff
@@ -1 +1,102 @@
...
+### Set HyperFormula version for all demos
+
+This command updates the `package.json` file in all demos to use version *1.2.3* of hyperformula package:
+
+```bash
+# in hyperformula-demos directory
+sh set-hyperformula-version.sh 1.2.3
+```
+
+### Test demos with unpublish version of HyperFormula
```
Comment (warpech) [inline README.md]: I think the whole title could be renamed to `Testing the demos with an unpublished version of HyperFormula`

File: README.md [hunk 5]
```diff
@@ -1 +1,102 @@
...
+
+To achive this a private package registry (such as [verdaccio](https://verdaccio.org/)) is necessary.
```
Comment (warpech) [inline README.md]: achive -> achieve

File: README.md [hunk 6]
```diff
@@ -1 +1,102 @@
...
+
+*We recommend using Node v16 to run the demos.*
+
+```bash
+# in hyperformula-demos directory
+cd basic-usage # or any other demo
+npm install
+npm run start
+# for most demos the browser opens automatically with the right URL
+# if it didn't, copy the URL from the console output and paste it into the browser
```
Comment (warpech) [inline README.md]: This would be more readable as a text after the code snippet, not within the snippet.

**Review by warpech — CHANGES_REQUESTED** [review]
Looking good! I suggested correcting some typos. I verified that all demos that have a `build` script, also have a `start` script.

File: README.md (hunk 1 above)
Comment (sequba) [reply README.md]: done

File: README.md (hunk 3 above)
Comment (sequba) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (sequba) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (sequba) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (sequba) [reply README.md]: You're right. Changed.

File: README.md (hunk 4 above)
Comment (sequba) [reply README.md]: done

File: README.md [hunk 7]
```diff
@@ -1 +1,102 @@
...
+
+The bugfixes for the previous versions are applied to the appropriate branches.
+
+### Updating HyperFormula version for all demos
+
+This command updates the `package.json` file in all demos to use version *x.y.z* of hyperformula package:
+
+```bash
+# in hyperformula-demos directory
+sh set-hyperformula-version.sh x.y.z
```
Comment (jansiegel) [inline README.md:70]: @sequba 
When running this command (on MacOS 10.15) I get this error:

```
sed: 1: "./react-demo/package.json": invalid command code .
```

Not sure if I'm doing something wrong?

Comment (sequba) [reply README.md:70]: It turns out MacOS uses a non-standard implementation of `sed`. I changed the script, so it should work for both GNU and MacOS versions of `sed`. @jansiegel Can you verify that on your system?

**Review by warpech — APPROVED** [review]
I approve my part of the review, but please make sure it works on MacOS

File: README.md (hunk 7 above)
Comment (warpech) [reply README.md:70]: The new code (2baacddf958048b9e3e959f0afe2f6db4e22c1a6) gives me another error:

```
sed: 1: "s/(^\s*"hyperformula":  ...": RE error: repetition-operator operand invalid
```

This seems to work:

```
find . -maxdepth 2 -name package.json | xargs sed -i '' 's/"hyperformula": "[^"]*"/"hyperformula": "'$1'"/g'
```

Comment (sequba) [reply README.md:70]: @warpech your version doesn't work with GNU sed. I commited another version:
```
find . -maxdepth 2 -name package.json -print0 | xargs -0 sed -i'' 's/"hyperformula": "[^"]*"/"hyperformula": "'$1'"/g'
```
@warpech, @jansiegel Can you try this one? I don't have access to MacOS PC to test it myself.

Comment (warpech) [reply README.md:70]: Unfortunately, this gives me an error:

```
sed: 1: "./react-demo/package.json": invalid command code .
```

The problem is in the lack of space after `sed -i`. The following command works on macOS:

```
find . -maxdepth 2 -name package.json -print0 | xargs -0 sed -i '' 's/"hyperformula": "[^"]*"/"hyperformula": "'$1'"/g'
```

Comment (warpech) [reply README.md:70]: After [03f00aa](https://github.com/handsontable/hyperformula-demos/pull/7/commits/03f00aab0669969c2a085703608e1184be98d502), script `set-hyperformula-version.sh` works ok on my macOS. Thanks!

**Review by warpech — APPROVED** [review]
After [03f00aa](https://github.com/handsontable/hyperformula-demos/pull/7/commits/03f00aab0669969c2a085703608e1184be98d502), script `set-hyperformula-version.sh` works ok on my macOS. Thanks!

**Pull request merged**